### ISSUE TITLE
Something wrong on this demo:AttributeError:'CamLoader_Q' object has no attribute 'ret'

### ISSUE THREAD

**Issue by RhInEs-liao** [GajuuzZ/Human-Falling-Detect-Tracks#56]
![sunlogin_20211003002044](https://user-images.githubusercontent.com/55306204/135724677-3bb41e6e-e05a-4e2f-a389-4f042a3dd63d.png)
AttributeError:'CamLoader_Q' object has no attribute 'ret'

I just change the source to my demo.mp4,if I dont change source,that will not identify the camera,I use yolov3~yolov5 that all can identify my webcam;What's wrong?!

my conda env {yolov5} setting: 
pytorch==1.8.0+cu111
CUDA==11.1
python==3.7
my device:
Lenovo Y7000 i5-9300H GTX1660ti

**Comment by GajuuzZ** [comment]
Ok, I fixed that. Try to run just only CameraLoader.py if you want to find that program can read your camera. Maybe you need to change the source number.

**Comment by RhInEs-liao** [comment]
OK，I just change ur commit，then I successfully open the frame of my webcam！but ，when I run main.py，there are some new problems on it；
![MV)M2SQG$9EH56B3RC{8B8K](https://user-images.githubusercontent.com/55306204/135738690-0131b32a-baa2-4e04-b337-9781ef760eb5.png)

**Comment by RhInEs-liao** [comment]
the first one code that load the '1s' video,after I run the code,then show a frame,and then show this error to me.

What's more, What is the meaning of "number of source"? 1 means camera? 2 means video file?

**Comment by RhInEs-liao** [comment]
I want to know that this demo how to train my own dataset?I see that there are some .py to make a datatest then train that

**Comment by RhInEs-liao** [comment]
new problems,I test a video with 3 minutes and 30 seconds,but I just get a 19s detect video result .After 20s,it quickly shut down,and any error is not exist,common finished.I am puzzled;

**Comment by mufenzhimi** [comment]
> OK，I just change ur commit，then I successfully open the frame of my webcam！but ，when I run main.py，there are some new problems on it； ![MV)M2SQG$9EH56B3RC{8B8K](https://user-images.githubusercontent.com/55306204/135738690-0131b32a-baa2-4e04-b337-9781ef760eb5.png)

How did you fix this problem? It seems that I have the same problem.

**Comment by RhInEs-liao** [comment]
do you have some other community APPs？I can add you to talk about this demo

**Comment by RhInEs-liao** [comment]
The OMP problem，I just add two lines code to solve it；
you can try to add this code to the begin of your "main.py"
import os
os.environ[“KMP_DUPLICATE_LIB_OK”]=“TRUE”

**Comment by RhInEs-liao** [comment]
> > OK，I just change ur commit，then I successfully open the frame of my webcam！but ，when I run main.py，there are some new problems on it； ![MV)M2SQG$9EH56B3RC{8B8K](https://user-images.githubusercontent.com/55306204/135738690-0131b32a-baa2-4e04-b337-9781ef760eb5.png)
> 
> How did you fix this problem? It seems that I have the same problem.

The OMP problem，I just add two lines code to solve it；
you can try to add this code to the begin of your "main.py"
import os
os.environ[“KMP_DUPLICATE_LIB_OK”]=“TRUE”

**Comment by RhInEs-liao** [comment]
> > OK，I just change ur commit，then I successfully open the frame of my webcam！but ，when I run main.py，there are some new problems on it； ![MV)M2SQG$9EH56B3RC{8B8K](https://user-images.githubusercontent.com/55306204/135738690-0131b32a-baa2-4e04-b337-9781ef760eb5.png)
> 
> How did you fix this problem? It seems that I have the same problem.

you can add my "Tencent QQ":1374407026

**Comment by mufenzhimi** [comment]
OKK!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "GajuuzZ/Human-Falling-Detect-Tracks"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年10月3日(星期天) 晚上10:12
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [GajuuzZ/Human-Falling-Detect-Tracks] Something wrong on this demo:AttributeError:'CamLoader_Q' object has no attribute 'ret' (#56)





   
OK，I just change ur commit，then I successfully open the frame of my webcam！but ，when I run main.py，there are some new problems on it； 
  
How did you fix this problem? It seems that I have the same problem.
  
you can add my "Tencent QQ":1374407026
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by GajuuzZ** [comment]
Camera source number is the same parameter when you use “cv2.VideoCapture”.

**Comment by RhInEs-liao** [comment]
> Camera source number is the same parameter when you use “cv2.VideoCapture”.

NO.1     new problems,I test a video with 3 minutes and 30 seconds,but I just get a 19s detect video result .After 20s,it quickly shut down,and any error is not exist,common finished.I am puzzled;
NO.2    I want to know that this demo how to train my own dataset?I see that there are some .py to make a datatest then train that
NO.3    do you have some other community APPs？I can add you to talk about this demo;
@GajuuzZ

**Comment by RhInEs-liao** [comment]
@GajuuzZ

**Comment by GajuuzZ** [comment]
1. Maybe my CameraLoader is not that perfect. You can switch between two classes as i comment in the class description or use the traditional way. I will see the problem later when I have time.
2. This is not a tutorial project for beginners. Just for the guideline. I already provide that stuff as much as I can for the training. If you want to know you need to hack it by yourself.
3. I can't help you that much. I have my things to do too.

**Comment by WangJi1022** [comment]
> > 当您使用“cv2.VideoCapture”时，摄像机源编号是相同的参数。
> 
> NO.1 新问题，我测试了一个3分30秒的视频，但是我只是得到一个19s的检测视频结果。20s后，它很快就关机了，没有任何错误存在，常见的结束了。我很困惑； NO.2 我想知道这个demo怎么训练我自己的数据集？我看到有一些.py做一个datatest然后训练 NO.3你还有其他一些社区APP吗？我可以加你谈谈这个演示； @GajuuzZ
我也遇到这个问题，为啥呢，而且似乎会跳帧

**Comment by af-doom** [comment]
> new problems,I test a video with 3 minutes and 30 seconds,but I just get a 19s detect video result .After 20s,it quickly shut down,and any error is not exist,common finished.I am puzzled;

Have you solved this problem?

**Comment by heavenluv** [comment]
> > new problems,I test a video with 3 minutes and 30 seconds,but I just get a 19s detect video result .After 20s,it quickly shut down,and any error is not exist,common finished.I am puzzled;

This one is due to the buffer queue size not big enough. If your video is long then you must choose the larger buffer for your CamLoader_Q.

**Comment by tuchengfeng** [comment]
> OKK!
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "GajuuzZ/Human-Falling-Detect-Tracks" ***@***.***&gt;; 发送时间:&nbsp;2021年10月3日(星期天) 晚上10:12 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [GajuuzZ/Human-Falling-Detect-Tracks] Something wrong on this demo:AttributeError:'CamLoader_Q' object has no attribute 'ret' (#56) OK，I just change ur commit，then I successfully open the frame of my webcam！but ，when I run main.py，there are some new problems on it； How did you fix this problem? It seems that I have the same problem. you can add my "Tencent QQ":1374407026 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

![image](https://user-images.githubusercontent.com/61649382/163785059-1cb7e386-fb7a-40b1-9f9f-675f2fedeac3.png)
was your problem solved？how u solved it?like this?
